### PR TITLE
fix(27041): Align backend and frontend on bi-directionality

### PR DESCRIPTION
**Pull request by vanch3d**
See https://hivemq.kanbanize.com/ctrl_board/57/cards/27041/details/

This PR is part of the bi-directional epic and aligns the content of the backend with the frontend. It removes a significant layer of mocks in order to interact with the live API endpoints properly. 

The PR fixes a few bugs created during the migration from subscription to mappings. It refactors the use of the `RSJ` form across the Edge code and fixes discrepancies between different templates. 

The PR removes HTML validation from the JSONSchema-based form (see [https://rjsf-team.github.io/react-jsonschema-form/docs/usage/validation/#html5-validation
      noHtml5Validate ](https://rjsf-team.github.io/react-jsonschema-form/docs/usage/validation/#html5-validation) and introduces a hack to deal with unloadable meta-schema, see https://hivemq.kanbanize.com/ctrl_board/57/cards/27657/details/ 

The PR also updates some of the dependencies to address known vulnerabilities (closes #605)

### Out-of-scope
- The ticket only supports inward and outward mappings. Tags and Topic filters will be dealt with in a subsequent ticket

### Before 
screenshot-localhost_3000-2024_11_04-12_00_52

screenshot-localhost_3000-2024_11_04-12_00_25

### After.
screenshot-localhost_3000-2024_11_04-12_03_25

screenshot-localhost_3000-2024_11_04-12_04_00

screenshot-localhost_3000-2024_11_04-12_04_15

screenshot-localhost_3000-2024_11_04-12_04_26

### CODE REVIEW

File: hivemq-edge/src/frontend/src/components/rjsf/Templates/DescriptionFieldTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  // Override to fix bug with nested p
+  if (typeof description === 'string') {
+    return (
+      <Text id={id} mt={2} mb={4}>
+        {description}
+      </Text>
+    )
+  }
+
+  return <>{description}</>
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/components/rjsf/Templates/DescriptionFieldTemplate.tsx:26]: suggestion(non-blocking): can be reduced to:

```suggestion
  return description
```

File: hivemq-edge/src/frontend/src/modules/ProtocolAdapters/components/drawers/AdapterInstanceDrawer.tsx [hunk 2]
```diff
@@ -67,62 +56,43 @@ const AdapterInstanceDrawer: FC<AdapterInstanceDrawerProps> = ({
...
     const adapter: ProtocolAdapter | undefined = data?.items?.find((e) => e.id === adapterType)
     const { configSchema, uiSchema, capabilities } = adapter || {}
 
-    // TODO[NVL] This is still a hack; backend needs to provide identification of mapping-related properties
-    const paths = getTopicPaths(configSchema || {})
-    const subIndex = getMainRootFromPath(paths)
-    const hiddenMappingsKey = import.meta.env.VITE_FLAG_ADAPTER_SPLIT_SCHEMA === 'true' ? subIndex : undefined
+    // If the config schema is split across entities, replace the fields by a warning
+    const hideProperties = []
+    if (import.meta.env.VITE_FLAG_ADAPTER_MAPPINGS_IN_WORKSPACE === 'true' && adapter?.id) {
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/modules/ProtocolAdapters/components/drawers/AdapterInstanceDrawer.tsx:66]: question: I've seen this one multiple times maybe worth extracting:

some file

```ts
export const FLAG_ADAPTER_MAPPINGS_IN_WORKSPACE = import.meta.env.VITE_FLAG_ADAPTER_MAPPINGS_IN_WORKSPACE === 'true'
```

`FLAG_ADAPTER_MAPPINGS_IN_WORKSPACE`

**Review by h2xd — APPROVED** [review]
LGTM! 🐊

File: hivemq-edge/src/frontend/src/modules/ProtocolAdapters/components/drawers/AdapterInstanceDrawer.tsx (hunk 2 above)
Comment (vanch3d) [reply hivemq-edge/src/frontend/src/modules/ProtocolAdapters/components/drawers/AdapterInstanceDrawer.tsx:66]: Nice nudge! I do have a mechanism in place for these... and I forgot about it!
https://github.com/hivemq/hivemq-edge/blob/master/hivemq-edge/src/frontend/src/config/index.ts

**Pull request merged**